bitnami/external-dns/templates/_helpers.tpl

```diff
@@ -428,3 +428,14 @@ external-dns: transip.apiKey
     Please set the apiKey parameter (--set transip.apiKey="xxxx")
 {{- end -}}
 {{- end -}}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "serviceaccount.name" -}}
+{{- if and .Values.rbac.create .Values.rbac.serviceAccount.create -}}
+{{ include "external-dns.fullname" . }}
+{{- else -}}
+{{ .Values.rbac.serviceAccountName }}
+{{- end -}}
+{{- end -}}
```

Comment:
Suggestion: you just need to check if `.Values.rbac.serviceAccountName` is defined. `.Values.rbac.serviceAccount.create` doesn't add any value.
```
{/*
Return the ExternalDNS service account name
*/}}
{{- define "external-dns.serviceAccountName" -}}
{{- if .Values.rbac.serviceAccountName -}}
    {{- printf "%s" (tpl .Values.rbac.serviceAccountName $) -}}
{{- else -}}
    {{- printf "%s" (include "external-dns.fullname" .) -}}
{{- end -}}
{{- end -}}
```